XenonCode.hpp

```diff
@@ -7326,6 +7331,14 @@ const int VERSION_PATCH = 0;
 										 throw RuntimeError("TODO this type for self recursion");
 									}
 								} break;
+								case HSH: {// REF_DST REF_SRC
+									ByteCode dst = nextCode();
+									ByteCode val = nextCode();
+									if (IsNumeric(dst) && IsText(val)) {
+										std::string str = MemGetText(val);
+										MemSet((double)((int64_t)(std::hash<std::string>{}(str)) & ((1l<<53)-1)), dst);
```

Comment:
Potential issue here is that std::hash::operator() returns an unsigned 64 bit integer, whereas with XenonCode VM we store numeric values as FP64 so the value would lose precision and the hash values distribution won't be uniform... I would simply add here a modulus 2^52 to only use the mantissa of FP64 to bring it back to a uniform distribution and it will also make it safe to compare with other values in XC if needed.
Other than that, it's all good, I could merge it in about a week from now.

Reply:
Just noticed you pushed a second commit with pretty much what I was asking for here.
I will however add that in Windows you need to use long long in your literals, so 1ll<<53
Otherwise it will truncate it to 32 bit (since a long int is 32 bit in windows gcc)

Reply:
Ah, good catch [b7ea764](https://github.com/batcholi/XenonCode/pull/5/commits/b7ea76477845effd8c296f5f56e87ae8597d3ba4) changes it to `1ll`